src/vs/workbench/contrib/files/common/explorerService.ts

```diff
@@ -159,6 +159,12 @@ export class ExplorerService implements IExplorerService {
 		const options: IResolveFileOptions = { resolveTo: [resource], resolveMetadata: this.sortOrder === 'modified' };
 		const workspaceFolder = this.contextService.getWorkspaceFolder(resource);
 		const rootUri = workspaceFolder ? workspaceFolder.uri : this.roots[0].resource;
+
```

Comment:
Your check makes sense but I think we can make this more general.
I do not like the statment on line 161 that if the workspaceFolder does not exist we take the first root.

Let's change that, suche that if the workspaceFolder is not found the rootUri is undefined.
And if it is undefined we should just return.

I think this should also handle your use case. Let me know what you think about this and how ti goes.

Reply:
@isidorn I agree. Have just updated my branch.